ui/src/features/Editor/components/Canvas/useNodes.ts

```
@@ -154,38 +156,54 @@
-              droppedNode.handles?.find((h) => h.type === "source")?.type ??
-              null,
-            target: e.target,
-            targetHandle: e.targetHandle ?? null,
-          };
-          newEdges = addEdge(newConnectionB, newEdges);
-
-          onEdgesChange(newEdges);
+          const removeChanges: EdgeChange[] = edges.reduce((acc, edge) => {
+            if (edge.source === e.source && edge.target === e.target) {
+              return [
+                ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)